src/Evolution/Systems/ScalarWave/Constraints.hpp

```diff
@@ -0,0 +1,109 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+
+/// \cond
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace ScalarWave {
+// @{
+/*!
+ * \brief Computes the scalar-wave one-index constraint.
+ *
+ * \details Computes the scalar-wave one-index constraint,
+ * \f$C_{i} = \partial_i\psi - \Phi_{i},\f$ which is
+ * given by Eq. (19) of \cite Holst2004wt
+ */
+template <size_t SpatialDim>
+tnsr::i<DataVector, SpatialDim, Frame::Inertial> one_index_constraint(
+    const tnsr::i<DataVector, SpatialDim, Frame::Inertial>& d_psi,
+    const tnsr::i<DataVector, SpatialDim, Frame::Inertial>& phi) noexcept;
+
+template <size_t SpatialDim>
+void one_index_constraint(
+    gsl::not_null<tnsr::i<DataVector, SpatialDim, Frame::Inertial>*> constraint,
+    const tnsr::i<DataVector, SpatialDim, Frame::Inertial>& d_psi,
+    const tnsr::i<DataVector, SpatialDim, Frame::Inertial>& phi) noexcept;
+// @}
+
+// @{
+/*!
+ * \brief Computes the scalar-wave 2-index constraint.
+ *
+ * \details Computes the scalar-wave 2-index FOSH constraint that
```

Comment:
write out the abbreviation `FOSH`

delete `that`

Reply:
rephrased